### PR TITLE
Move first run and whats new to onboarding package

### PR DESCRIPTION
**Pull request by ezaquarii**
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

**Comment by ezaquarii** [comment]
This is a proposition to rework currently separate `WhatsNew` and `FirstRun` concepts into an "user onboarding" package, responsible for any first-run-popup, watch-our-new-things-messagaging, we-want-your-attention-here-and-now scenarios.

Also, more `AccountUtils` API calls moved to `UserAccountManager`.

This change brings some more Kotlin examples in application code, to test our desire to move deeper into that territory.

I marked this PR as review ready to get some feedback, but it's still under development and I'd like to pause here a bit for reflections until receiving more feedback from you.

**Comment by ezaquarii** [comment]
xref #3884

**Comment by AndyScherzinger** [comment]
I do like the change and I am quite happy that I can actually read and understand it right away :)

What I can't say though if if the SpotBugs score improved simply by code now being kotlin and not java anymore and thus the analyzer not finding issues anymore? Do you know this @ezaquarii ?

**Comment by ezaquarii** [comment]
@AndyScherzinger @tobiasKaminsky 
> What I can't say though if if the SpotBugs score improved simply by code now being kotlin and not java anymore and thus the analyzer not finding issues anymore? Do you know this @ezaquarii ?

There are few factors here:
1) `spotbugs` works for Java only - you are right about this
2) those refactorings we're doing generally break cyclic dependencies that `spotbugs` complains about quite a lot

We need to adopt `ktlint` and `detekt` to handle that, but so far we have no Kotlin code apart this one  little class and 1 test, so it wasn't in rush yet. However, this is pretty well understood process and both tools are solid, so I don't expect any problems with it.

**Comment by ezaquarii** [comment]
I diffed the reports and it looks like in this case there are some improvements in `com.owncloud.android.datamodel`... which I didn't touch at all.

Without investigating the bytecode I can only speculate why this happened.

**Comment by Unpublished** [comment]
@ezaquarii Fixed the failing test (caused by rebase) for you, can be squashed before merge.

File: src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 1]
```diff
@@ -57,6 +57,7 @@
 import com.google.android.material.navigation.NavigationView;
 import com.nextcloud.client.account.UserAccountManager;
 import com.nextcloud.client.di.Injectable;
+import com.nextcloud.client.onboarding.FirstRunActivity;
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: Strange that Codacy does not complain…

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: Hm. Ok. It is needed, but why did it then not failed previously?

**Comment by tobiasKaminsky** [comment]
> We need to adopt `ktlint` and `detekt` to handle that

I'll give it a try in a new PR, so we keep tracking of those warnings.

**Comment by AndyScherzinger** [comment]
GPlay build fails:
```

> Task :compileGplayDebugAndroidTestJavaWithJavac FAILED
--
557 | /drone/src/src/androidTest/java/com/owncloud/android/ui/activity/FileDisplayActivityTest.java:5: error: package com.********.client.whatsnew does not exist
558 | import com.********.client.whatsnew.WhatsNewActivity;
559 | ^
560 | /drone/src/src/androidTest/java/com/owncloud/android/ui/activity/FileDisplayActivityTest.java:29: error: cannot find symbol
561 | if (activity instanceof WhatsNewActivity) {
562 | ^
563 | symbol: class WhatsNewActivity
564 | location: class FileDisplayActivityTest


```

**Comment by AndyScherzinger** [comment]
@ezaquarii the build has a compile issue, see my Comment above

**Comment by ezaquarii** [comment]
> @ezaquarii the build has a compile issue, see my Comment above

No worries, I'll make the branch straight.

**Comment by ezaquarii** [comment]
@tobiasKaminsky  Codacy failed again with no reason. I squashed the branch and force-pushed to force rebuild - it seem to have some effect on the other one.

**Comment by ezaquarii** [comment]
@tobiasKaminsky 
```
Task 'createGplayDebugCoverageReport' not found in root project 'src'.
```

**Comment by ezaquarii** [comment]
@tobiasKaminsky Could you pls take a careful look at `analysis-wrapper.sh`? There were some conflicts and I'm not sure I resolved them properly.

File: build.gradle [hunk 2]
```diff
@@ -368,7 +374,7 @@ task ktlint(type: JavaExec, group: "verification") {
     description = "Check Kotlin code style."
     main = "com.pinterest.ktlint.Main"
     classpath = configurations.ktlint
-    args "--reporter=plain,output=${buildDir}/ktlint.txt,src/**/*.kt"
+    args "--reporter=plain", "--reporter=plain,output=${buildDir}/ktlint.txt,src/**/*.kt"
```
Comment (ezaquarii) [inline build.gradle]: Also prints issues to stdout, so IDE provides clickable links to errors.

**Comment by AndyScherzinger** [comment]
pinging @tobiasKaminsky for review/merge ❤️ All checks ✅

**Pull request merged**